src/AudioInputI2S.h

```diff
@@ -0,0 +1,45 @@
+/*
+  AudioInputI2S
+  I2S audio source
+
+*/
+
+#ifndef _AUDIOINPUTI2S_H
+#define _AUDIOINPUTI2S_H
+
+#include <Arduino.h>
+#include "AudioGenerator.h"
+
+class AudioInputI2S : public AudioGenerator
+{
+  public:
+    AudioInputI2S(int port=0, int dma_buf_count = 8, int use_apll=APLL_DISABLE);
+    virtual ~AudioInputI2S() override;
+    bool SetPinout(int bclkPin, int wclkPin, int dinPin);
+    virtual bool begin(AudioFileSource *source, AudioOutput *output) override;
+    virtual bool loop() override;
+    virtual bool stop() override;
+    virtual bool isRunning() override;
+    virtual bool SetRate(int hz);
+    virtual bool SetBitsPerSample(int bits);
+    virtual bool SetGain(float f);
+    uint32_t GetSample(void);
+    virtual uint32_t read(void* data, size_t len_bytes);
```

Comment:
Does this or GetSample() need to be exposed to the end user at all?  If it's an AudioGenerator, then the output stage gets the data fed to it as it becomes available.  If you end up doing read() from your app code, it may be simpler just to call the ESP SDK directly, no?